js/iframe-coordinator/libs/client.ts

```diff
@@ -3,70 +3,110 @@ import { PublicationHandler } from "./types";
 
 interface ToastOptions {
   title?: string;
-  custom?: { [x: string]: any };
+  custom?: any;
+}
+
+interface ClientConfigOptions {
+  clientWindow?: Window
 }
 
 class Client {
   private _worker: ClientProgram;
+  private _isStarted: boolean;
+  private _clientWindow: Window;
   private _messageHandlers: Array<PublicationHandler> = [];
 
-  start(): void {
-    if (!this._worker) {
-      this._worker = Elm.Client.init();
-  
-      window.addEventListener("message", (event: MessageEvent) => {
-        this._worker.ports.fromHost.send({
-          origin: event.origin,
-          data: event.data
-        });
-      });
-  
-      this._worker.ports.toHost.subscribe((message: any) => {
-        window.parent.postMessage(message, "*");
-      });
-  
-      this._worker.ports.toClient.subscribe((message: any) => {
-        if (message.msgType == "publish") {
-          this._messageHandlers.forEach(handler => {
-            handler(message.msg);
-          });
-        }
-      });
+  public constructor(configOptions: ClientConfigOptions = {}) {
+    this._clientWindow = (configOptions.clientWindow) || window;
+    this._worker = Elm.Client.init();
+  }
+
+
+  private _sendingMessageToHost = (message: LabeledMsg) => {
```

Comment:
This is annoying part about js/ts classes.  Methods don't bind this automatically.  So instead we use private fields which do with lambdas, so that we can add/remove the same instance.

I think this is the best of the available options, but I am also ok with making these methods again and having fields for the bound versions of the methods.  (Or suggest something else).